### PR TITLE
Bug fix for issue #473 : SvgText wrong position in some scenario

### PR DESCRIPTION
**Pull request by francois-dorin**
Pull request for fix the bug #473.

I proceed by changing the base class of `SvgUnitCollection` from `List<SvgUnit>` to `ObservableCollection<SvgUnit>`. I made the needed changes and wrote a unit test.

This PR may cause some compatibility issue if methods specific to List were used (like AddRange, but I provided an implementation to minimize the impact).

I had to add a reference to an assembly:
- WindowsBase.dll for Net35
- Nuget package "System.ObjectModel" for other targets

I do not test each target platform (I do not know how to do it), but unit test and SvgW3CTestRunner run without any problem.

### CODE REVIEW

File: Source/Svg.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.2;net35</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.2;net35;net452;net472</TargetFrameworks>
```
Comment (mrbean-bremen) [inline Source/Svg.csproj]: Why you needed to add these? The 3.5 version should be compatible with later versions.

File: Source/Svg.csproj [hunk 2]
```diff
@@ -37,6 +37,7 @@ Supports multiple targets v3.5 thru to dotnetcore2.2.
 NetStandard does not fully support the Drawing2D package - so has been left out.
     </PackageReleaseNotes>
     <PackageProjectUrl>https://github.com/vvvv/SVG</PackageProjectUrl>
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
```
Comment (mrbean-bremen) [inline Source/Svg.csproj]: Has this been added intentionally?

File: Source/Basic Shapes/SvgVisualElement.cs [hunk 3]
```diff
@@ -288,7 +288,7 @@ protected internal virtual bool RenderStroke(ISvgRenderer renderer)
                                     }
 
                                     /* divide by stroke width - GDI behaviour that I don't quite understand yet.*/
-                                    pen.DashPattern = this.StrokeDashArray.ConvertAll(u => ((u.ToDeviceValue(renderer, UnitRenderingType.Other, this) <= 0) ? 1 : u.ToDeviceValue(renderer, UnitRenderingType.Other, this)) /
+                                    pen.DashPattern = this.StrokeDashArray.Select(u => ((u.ToDeviceValue(renderer, UnitRenderingType.Other, this) <= 0) ? 1 : u.ToDeviceValue(renderer, UnitRenderingType.Other, this)) /
```
Comment (mrbean-bremen) [inline Source/Basic Shapes/SvgVisualElement.cs]: Is this related to the fix?

**Comment by mrbean-bremen** [comment]
Thanks for the PR! I added a couple of questions, and I noticed that one of the image comparison tests failed (painting-stroke-06-t.svg). Can you please check this?

File: Source/Svg.csproj (hunk 1 above)
Comment (francois-dorin) [reply Source/Svg.csproj]: ObservableCollections is not in the same assembly depending of the targetted framework. With the 3.5 it is in WindowsBase.dll. In 4.x it is in System.ObjectModel.
Without adding these frameworks, an application targetting 4.7.2 for example need to explicitly references the WindowsBase.dll.

File: Source/Svg.csproj (hunk 2 above)
Comment (francois-dorin) [reply Source/Svg.csproj]: No. This one can be reverted.

File: Source/Basic Shapes/SvgVisualElement.cs (hunk 3 above)
Comment (francois-dorin) [reply Source/Basic Shapes/SvgVisualElement.cs]: Yes. ConvertAll is a method defined in List<T> but not in ObservableCollection, and do the same thing that Select (which works with any IEnumerable as it is a Linq method).

**Comment by francois-dorin** [comment]
I check the failing unit test. It seems not related to my PR as it also failed on commit 21d61ec "Create CONTRIBUTING.md"

File: Source/Basic Shapes/SvgVisualElement.cs (hunk 3 above)
Comment (mrbean-bremen) [reply Source/Basic Shapes/SvgVisualElement.cs]: Ah ok, thanks!

File: Source/Svg.csproj (hunk 1 above)
Comment (mrbean-bremen) [reply Source/Svg.csproj]: Ok, I see - makes sense.

**Comment by mrbean-bremen** [comment]
Ok, I see now - I hadn't noticed this, as it passed in master. Seems to be some transient problem with loading the file, I will check if I can understand it...

File: Source/Svg.csproj (hunk 1 above)
Comment (francois-dorin) [reply Source/Svg.csproj]: And to complete: as WindowsBase.dll is not in the Framework 4.x, to use WindowsBase.dll we have to provide a full path to the assembly! So, adding new target framework here seems a better approach to me.

Comment (mrbean-bremen) [reply Source/Svg.csproj]: Yes, I agree. I also just read up about `AutoGenerateBindingRedirects`, and I think it makes sense to let this switched on, so I will merge this as is.

**Comment by mrbean-bremen** [comment]
Hm, the test still fails in master, and it looks like it is related to the change - it crashes on these lines:
```c#
    if (this.StrokeDashArray.Count % 2 != 0)
    {
         // handle odd dash arrays by repeating them once
         this.StrokeDashArray.AddRange(this.StrokeDashArray);
     }
```
as `StrokeDashArray` is now not mutable. I wonder how it passed for you? It also crashes in the W3CTestRunner.

**Comment by francois-dorin** [comment]
Hmmm, it is a special case where the collection is added to itself! I will correct it

**Comment by mrbean-bremen** [comment]
Well, I can do that if you want - it is not a big change.

**Comment by mrbean-bremen** [comment]
I just tried it, and it works now.

**Comment by mrbean-bremen** [comment]
Ok, I checked it in - my responsibility now, as I have overlooked it in the review :)

**Comment by mrbean-bremen** [comment]
Looks like the other failing test in my own commit was due to a forced commit, so this was not related. I got confused... Anyway, the tests are now green - thanks again for the contribution!

**Comment by francois-dorin** [comment]
No problem. Thank you for correcting it ;)
